src/ui/pages/codi/zcl_abapgit_gui_page_whereused.clas.abap

```diff
@@ -0,0 +1,222 @@
+
+    IF mi_table IS BOUND.
+      RETURN.
+    ENDIF.
+
+    IF ls_sorting_state-column_id IS INITIAL.
+      ls_sorting_state-column_id = 'package'.
+    ENDIF.
+
+    mi_table = zcl_abapgit_html_table=>create(
+      ii_renderer              = me
+      is_initial_sorting_state = ls_sorting_state ).
+    mi_table->define_column_group(
+        iv_group_title = 'Repo object'
+        iv_group_id    = '' " No need
+      )->define_column(
+        iv_column_id    = 'dep_package'
+        iv_column_title = 'Pkg'
+      )->define_column(
+        iv_column_id    = 'dep_obj_type'
+        iv_column_title = 'Type'
+      )->define_column(
+        iv_column_id    = 'dep_obj_name'
+        iv_column_title = 'Obj name'
+      )->define_column(
+        iv_column_id    = 'dep_used_obj'
+        iv_column_title = 'Used obj' ).
+    mi_table->define_column_group(
+        iv_group_title = 'Used in'
+        iv_group_id    = 'where' " Needed for CSS
+      )->define_column(
+        iv_column_id    = 'package'
+        iv_column_title = 'Pkg'
+      )->define_column(
+        iv_column_id    = 'obj_type'
+        iv_column_title = 'Type'
+      )->define_column(
+        iv_column_id    = 'obj_name'
+        iv_column_title = 'Obj name' ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_gui_event_handler~on_event.
+
+    IF mi_table->process_sorting_request( ii_event->mv_action ) = abap_true.
+      rs_handled-state = zcl_abapgit_gui=>c_event_state-re_render.
+      RETURN.
+    ENDIF.
+
+    CASE ii_event->mv_action.
+      WHEN c_action-refresh.
+        rs_handled-state = zcl_abapgit_gui=>c_event_state-re_render.
+      WHEN OTHERS.
+        RETURN.
+    ENDCASE.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_gui_hotkeys~get_hotkey_actions.
+
+    DATA ls_hotkey_action LIKE LINE OF rt_hotkey_actions.
+
+    ls_hotkey_action-ui_component = c_title.
+
+    ls_hotkey_action-description = |Refresh|.
+    ls_hotkey_action-action = c_action-refresh.
+    ls_hotkey_action-hotkey = |r|.
+    INSERT ls_hotkey_action INTO TABLE rt_hotkey_actions.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_gui_menu_provider~get_menu.
+
+    ro_toolbar = zcl_abapgit_html_toolbar=>create( )->add(
+      iv_txt = 'Refresh'
+      iv_act = c_action-refresh ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_gui_page_title~get_page_title.
+    rv_title = c_title.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_gui_renderable~render.
+
+    DATA lt_where_used TYPE zcl_abapgit_where_used_tools=>ty_dependency_tt.
+
+    register_handlers( ).
+    init_table_component( ).
+
+    CREATE OBJECT ri_html TYPE zcl_abapgit_html.
+
+    ri_html->div(
+      iv_class   = 'wu-header'
+      iv_content = |Where used for package {
+        zcl_abapgit_gui_chunk_lib=>render_package_name( mv_package )->render( iv_no_line_breaks = abap_true )
+        } and it's subpackages| ).
+
+    lt_where_used = zcl_abapgit_where_used_tools=>new( )->select_external_usages( mv_package ).
```

Comment:
Should depend on "Ignore sub-package" setting

Reply:
This one ?

![image](https://github.com/abapGit/abapGit/assets/15635498/11d3a360-d512-4da4-8c22-30383bd6105e)

hmmm, that means requiring the full repo object in this page rather than just the package name ... Interesting. What 's the use of this option? I can imagine ignoring specific subpackages but all ... hmm. Let me think about it. There is something unnatural about this flag.

Reply:
And if sub-packages are omitted, should the usages of THE PACKAGE in THE SUBPACKAGES be displayed? Hmmm. very strange flag ... :)

Reply:
The flag existed "forever". It's useful if you have one repo per package, no matter the package hierarchy or if you want to serialize a package in the middle of the hierarchy.

Reply:
"forever" does not except that it is strange and suboptimal. The excluded code should be outside of the package or in a specific sub-package(s) but not "all" subpackages - this is a very weird and wrong case. This feature should be redesigned imho. But OK, it does is not applicable to this PR. I'll see how to fix it. It's just really not straight forward especially in terms of displaying connections with exclusions

Reply:
`lo_repo->get_tadir_objects( )` takes all settings into consideration and gives you the right objects